packages/scene-composer/src/components/StateManager.tsx

```diff
@@ -184,6 +188,15 @@ const StateManager: React.FC<SceneComposerInternalProps> = ({
             : entityBindingComponent?.valueDataBinding.dataBindingContext,
         });
       }
+      if (dataOverlayComponent) {
```

Comment:
I remember made comment previously related to this part that we shouldn't push additionalComponentData in this hard coded order. the component data should match with the type of components in `componentTypes`. 

so we should loop through the `componentTypes` array, and foreach type, check its type to see what additional data to be pushed. for type that doesn't have additional data, we should either push `undefined` or empty at that index so the `componentTypes` and `additionalComponentData` can match

Reply:
Yeah now that we have more than 2. I think that should be a quick change.